field/FieldCreate.F90

```diff
@@ -90,6 +90,7 @@ subroutine field_empty_complete( field, &
       type(ESMF_Geom) :: geom
       integer :: dim_count, idim, status
 
+      _ASSERT(present(num_levels) .eqv. present(vert_staggerloc), "num_levels and vert_staggerloc must be both present or both absent")
```

Comment:
Screams for a derived type that encapsulates both.  But ... probably not workable.

Reply:
Reverting the commit due to other issues. Will do this as a separate PR.